src/middle/Expr.mli

```diff
@@ -0,0 +1,104 @@
+open Core_kernel
+open Common
+
+type litType = Mir_pattern.litType = Int | Real | Str
+[@@deriving sexp, hash, compare]
+
+type 'a index = 'a Mir_pattern.index =
+  | All
+  | Single of 'a
+  | Upfrom of 'a
+  | Between of 'a * 'a
+  | MultiIndex of 'a
```

Comment:
To try to help line counts, I'd propose just not having .mli files where you want to expose the entire types underneath. That's how Mir.ml works now - almost everything in there is (supposed to be) a type so it doesn't need  a separate .mli file.